src/leihs/my/password_restore/back.clj

```diff
@@ -90,108 +88,118 @@
     (merge (anti-csrf-props request) <>)
     (merge {:footer {:appVersion release/version}} <>)))
 
+(defn user-with-unique-id-new [tx user-unique-id]
+  (let [sql {:select :* :from :users :where [:or [:= :users.org_id user-unique-id]
+                                             [:= :users.login user-unique-id]
+                                             [:= [[[:lower :users/email]]] [:lower (-> user-unique-id (or ""))]]
+                                             ]}
+        query (sql-format sql)]
+    (jdbc/execute-one! tx (spy query))))
+
 (defn forgot-get
   [request]
   (let [user-param (-> request
-                       :params
-                       :user)
-        tx (:tx request)
-        user (user-with-unique-id tx user-param)]
+                     :params
+                     :user)
+        tx (:tx-next request)
+        user (user-with-unique-id-new tx user-param)]
     (cond
       (-> request :settings :email_sending_enabled not)
       {:status 422,
        :headers {"Content-Type" "text/html"},
        :body (-> request
-                 common-props
-                 (merge {:userParam user-param,
-                         :flashMessages [{:messageID "password_forgot_email_sending_disabled_text"
-                                          :level "error"}]})
-                 auth-page)}
+               common-props
+               (merge {:userParam user-param,
+                       :flashMessages [{:messageID "password_forgot_email_sending_disabled_text"
+                                        :level "error"}]})
+               auth-page)}
 
       (-> user :email presence not)
       {:status 422,
        :headers {"Content-Type" "text/html"},
        :body (-> request
-                 common-props
-                 (merge {:userParam user-param,
-                         :flashMessages [{:messageID "password_forgot_user_has_no_email_flash_text"
-                                          :level "error"}]})
-                 auth-page)}
+               common-props
+               (merge {:userParam user-param,
+                       :flashMessages [{:messageID "password_forgot_user_has_no_email_flash_text"
+                                        :level "error"}]})
+               auth-page)}
 
       :else
       {:headers {"Content-Type" "text/html"}
        :body (-> request
-                 common-props
-                 (merge {:userParam user-param})
-                 auth-page)})))
-
-(defn forgot-post [{{user-param :user} :params tx :tx :as request}]
-  (let [user (user-with-unique-id tx user-param)
-        token (make-token 20)]
-    (if user
+               common-props
+               (merge {:userParam user-param})
+               auth-page)})))
+
+(defn forgot-post [{{user-param :user} :params tx :tx-next :as request}]
+  (let [user (user-with-unique-id-new tx (spy user-param))
+        token (make-token 20)
+        p (println ">>>token " token)
+        p (println ">>>user " user)
```

Comment:
avoid duplicate temp variables in `let` construct

`debug` instead of `println` or remove